docs/reference/contributing/guidelines/workflow.md

```diff
@@ -80,14 +95,28 @@ Each pull request goes through the following workflow:
 
 Labels that the Mbed OS maintainers add to a pull request represent the pull request workflow states. The Mbed OS maintainers are responsible for moving pull requests through the workflow states.
 
+Each state is time boxed. In most cases, this should be sufficient time to move to an another state.
+
 ##### Reviews
 
-All pull requests must be reviewed. The Mbed OS maintainers determine the most suitable person to review the pull request and tag that person accordingly. If a pull request requires a review from partners, the maintainers tag the corresponding GitHub group instead.
+All pull requests must be reviewed. The mbed CI bot determine the most suitable person to review the pull request and tag that person accordingly.
+
+Time: 3 days for reviewers to leave feedback since the label was added
 
 ##### The CI (Continuous Integration) testing
 
 There are a number of CI systems available. Which CI tests we run against a particular pull request depends on the effect of that pull request to the code base. Irrespective of which CIs tests run, Mbed OS has an all green policy, meaning that all the CI jobs that are triggered must pass before we merge the pull request.
 
+Time: 1 day for CI to complete and report back results
+
+The CI workflow can be found [here](tbd)
+
+##### Work needed
+
+The pull request requires additional work due to failed tests, changes requested.
+
+Time: 3 days to respond to the review comments
+
 ##### Releases
 
 Once we merge a pull request, we tag it with a release. This is the release in which we first publish this pull request. For patch releases, we allow only bug fixes, new targets and enhancements to existing functionality. New features only go to feature releases.
```

Comment:
On line 122, any note about feature branch PRs not getting a release tag?

Reply:
will add